src/components/Dmitry/Dmitry.tsx

```diff
@@ -1,45 +1,79 @@
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [checkedAdvanced, setCheckedAdvanced] = useState(false); // State variable to check advanced option.
+    const [checkedIPAddress, setCheckedIPaddress] = useState(false); // State variable to check IP address option.
+    const [checkedDomainname, setCheckedDomainname] = useState(false); // State variable to check domain name option.
+    const [checkedNetcraft, setCheckedNetcraft] = useState(false); // State variable to check Netcraft option.
+    const [checkedSubdomains, setCheckedSubdomains] = useState(false); // State variable to check subdomains option.
+    const [checkedEmailaddress, setCheckedEmailaddress] = useState(false); // State variable to check email address option.
+    const [checkedPortscan, setCheckedPortscan] = useState(false); // State variable to indicate if the port scan option is checked.
+    const [checkedFiltered, setCheckedFiltered] = useState(false); // State variable to indicate if the filtered option is checked.
+    const [checkedBanner, setCheckedBanner] = useState(false); // State variable to indicate if the banner option is checked.
+    const [delay, setDelay] = useState(2); // State variable to indicate delaying state in seconds.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
 
+    // Component Constants.
+    const title = "Deepmagic Information Gathering Tool"; // Title of the component.
+    const description =
+        "DMitry is a UNIX/(GNU) Linux command line application written in C. DMitry will scan to try and find " +
+        "possible subdomains, email addresses and uptime information, and further perform tcp port scans, and " +
+        "whois lookups."; // Description of the component.
+    const steps =
+        "Step 1: Enter a valid domain or IP address to be scanned.\n" +
+        "       Eg: 192.168.0.1\n\n" +
+        "Step 2: Enter a delay between requests. Default is 2 (milliseconds). Can be left blank.\n" +
+        "       Eg: 2\n\n" +
+        "Step 3: Click Start Scanning to commence Dmitry's operation.\n\n" +
+        "Step 4: View the Output block below to view the results of the tool's execution.\n";
+    const sourceLink = "https://www.kali.org/tools/dmitry/\n"; // Link to the source code (or Kali Tools).
```

Comment:
This link is broken. You do not need the \n. 
```suggestion
    const sourceLink = "https://www.kali.org/tools/dmitry/"; // Link to the source code (or Kali Tools).
```